engine/src/main/java/com/datatorrent/stram/StreamingContainerParent.java

```diff
@@ -208,20 +212,15 @@ public ContainerHeartbeatResponse processHeartbeat(ContainerHeartbeat msg)
     //LOG.debug("RPC latency from child container {} is {} ms (according to system clocks)", msg.getContainerId(),
     // now - msg.sentTms);
     dagManager.updateRPCLatency(msg.getContainerId(), now - msg.sentTms);
-    try {
-      final ContainerHeartbeat fmsg = msg;
-      return SecureExecutor.execute(new SecureExecutor.WorkLoad<ContainerHeartbeatResponse>()
+    final ContainerHeartbeat fmsg = msg;
+    return SecureExecutor.execute(new SecureExecutor.WorkLoad<ContainerHeartbeatResponse>()
+    {
+      @Override
+      public ContainerHeartbeatResponse run()
       {
-        @Override
-        public ContainerHeartbeatResponse run()
-        {
-          return dagManager.processHeartbeat(fmsg);
-        }
-      });
-    } catch (IOException ex) {
-      LOG.error("Error processing heartbeat", ex);
```

Comment:
That's because the recovery wasn't intended for internal errors but for connectivity issues including when the AM moves to new location. There is also an unnecessary assignment on line 215: final ContainerHeartbeat fmsg = msg; - can you also remove that while at it?